### ISSUE TITLE
Why do you zero the gradients after running the model?

### ISSUE THREAD

**Issue by vedantroy** [milesial/Pytorch-UNet#378]
Is there a reason to you call `optimizer.zero_grad(...)` *after* running the model?
I always thought the default training loop was something like this:

```
optimizer.zero_grad()
outputs = model(images)
predicted = outputs.round()
loss = criterion(outputs, masks)
loss.backward()
optimizer.step()     
```

**Comment by milesial** [comment]
Doing it before or after doesn't change anything, as long as gradients are zero before the backward happens
